js/compiler/compiler_test.go

```diff
@@ -203,5 +203,58 @@ func TestCorruptSourceMap(t *testing.T) {
 	msg, err := entries[0].String() // we need this in order to get the field error
 	require.NoError(t, err)
 
-	require.Contains(t, msg, `Could not load source map: missing \"mappings\" in sourcemap`)
+	require.Contains(t, msg, `Couldn't load source map for somefile`)
+	require.Contains(t, msg, `json: cannot unmarshal number into Go struct field v3.mappings of type string`)
+}
+
+func TestCorruptSourceMapOnlyForBabel(t *testing.T) {
+	t.Parallel()
+	// this a valid source map for the go implementation but babel doesn't like it
+	corruptSourceMap := []byte(`{"mappings": ";"}`)
```

Comment:
Fixing this will mean that we will be running babel possibly twice on every error it returns. 
I find it somewhat unlikely that if we have a json with `mappings` that it will not have the other  source map fields. Also hopefully we well really even less on babel soon.

Reply:
82a519b hopefully catches all the cases :crossed_fingers: